src/schedule/mod.rs

```diff
@@ -152,9 +149,9 @@ impl EpochSecrets {
 
     /// Remove the `encryption_secret` from the `EpochSecrets`, replacing it
     /// with `None` and return it.
-    pub fn remove_encryption_secret(&self) -> Result<Secret, KeyScheduleError> {
+    pub fn remove_encryption_secret(&mut self) -> Result<Secret, KeyScheduleError> {
```

Comment:
Maybe `consume_encryption_secret` is more appropriate, given that the secret is returned?